pkg/master/reconcilers/endpointsadapter_test.go

```diff
@@ -98,6 +108,28 @@ func TestEndpointsAdapterGet(t *testing.T) {
 			if !apiequality.Semantic.DeepEqual(endpoints, testCase.expectedEndpoints) {
 				t.Errorf("Expected endpoints: %v, got: %v", testCase.expectedEndpoints, endpoints)
 			}
+
+			epSliceList, err := client.DiscoveryV1alpha1().EndpointSlices(testCase.namespaceParam).List(metav1.ListOptions{})
```

Comment:
All of this test logic can actually be simplified because the EndpointSlice created here will always have a consistent name (nameParam here, but generally just `kubernetes`). I think it would make sense to test based on the existence of that EndpointSlice specifically with a `Get` call instead of a `List`.

Reply:
Makes sense. I have updated, PTAL.

Reply:
Hey @tnqn, this looks great, I think it could be simplified one more step to just use the deepEqual whether or not the expected value was nil. Something like what's done above for Endpoints: https://github.com/kubernetes/kubernetes/blob/master/pkg/master/reconcilers/endpointsadapter_test.go#L262-L264

Reply:
Sure, done.